packages/controls/src/common/types.ts

```diff
@@ -1,111 +1,26 @@
-  Width: 'Width',
-  Video: 'Video',
-} as const
-
-export const ControlDataTypeKey = '@@makeswift/type'
-
-export type Options<T> = T | ((props: Record<string, unknown>, deviceMode: Device) => T)
-
-export type ResolveOptions<T extends Options<unknown>> = T extends Options<infer U> ? U : never
-
-const deviceSchema = z.string()
-
-export type Device = z.infer<typeof deviceSchema>
-
-function createDeviceOverrideSchema<T extends z.ZodTypeAny>(
-  schema: T,
-): z.ZodObject<{ deviceId: typeof deviceSchema; value: T }> {
-  return z.object({
-    deviceId: deviceSchema,
-    value: schema,
-  })
-}
+export { type BoxModel } from 'css-box-model'
 
+export type Data = z.infer<typeof Schema.data>
+export type ColorData = z.infer<typeof Schema.colorData>
+export type ResolvedColorData = z.infer<typeof Schema.resolvedColorData>
+export type Device = z.infer<typeof Schema.deviceId>
 export type DeviceOverride<T> = { deviceId: Device; value: T }
-
-export function createResponsiveValueSchema<T extends z.ZodTypeAny>(
-  schema: T,
-): z.ZodArray<ReturnType<typeof createDeviceOverrideSchema<T>>> {
-  return z.array(createDeviceOverrideSchema(schema))
-}
-
 export type ResponsiveValue<T> = DeviceOverride<T>[]
+export type ResponsiveValueType<T> =
+  T extends ResponsiveValue<infer U> ? U : never
 
-export type ResponsiveValueType<T> = T extends ResponsiveValue<infer U> ? U : never
+export type ElementData = z.infer<typeof Schema.elementData>
+export type ElementReference = z.infer<typeof Schema.elementReference>
+export type Element = z.infer<typeof Schema.element>
 
-export const dataSchema: z.ZodType<Data> = z.any()
-
-export const elementDataSchema = z.object({
-  type: z.string(),
-  key: z.string(),
-  props: z.record(dataSchema),
-})
-
-export type ElementData = z.infer<typeof elementDataSchema>
-
-const elementReferenceSchema = z.object({
-  type: z.literal('reference'),
-  key: z.string(),
-  value: z.string(),
-})
-
-export type ElementReference = z.infer<typeof elementReferenceSchema>
-
-export const elementSchema = z.union([elementDataSchema, elementReferenceSchema])
-
-export type Element = z.infer<typeof elementSchema>
-
-export type TranslationDto = Record<string, Data>
-export type MergeTranslatableDataContext = {
-  translatedData: TranslationDto
-  mergeTranslatedData: (node: Element) => Element
-}
-
-export type ReplacementContext = {
-  elementHtmlIds: Set<string>
-  elementKeys: Map<string, string>
-  swatchIds: Map<string, string>
-  fileIds: Map<string, string>
-  typographyIds: Map<string, string>
-  tableIds: Map<string, string>
-  tableColumnIds: Map<string, string>
-  pageIds: Map<string, string>
-  globalElementIds: Map<string, string>
-  globalElementData: Map<string, ElementData>
+export function isElementReference(
+  element: Element,
+): element is ElementReference {
+  return !('props' in element)
 }
 
-export type CopyContext = {
-  replacementContext: ReplacementContext
-  copyElement: (node: Element) => Element
-}
```

Comment:
Moved contexts to their own file: https://github.com/makeswift/makeswift/pull/781/files#diff-69a50df54dce20530e7f5f578014e4c0f4d453ea327ff76393ae3d309b29f4ea